ui/app/AppLayouts/WalletV2/WalletHeader.qml

```diff
@@ -140,6 +229,479 @@ Item {
+                            pristine = true
+                            validationError = ""
+                            text = ""
+                        }
+
+                        label: qsTr("To")
+                        placeholderText: qsTr("Enter an ENS name or address")
+                        pasteFromClipboard: text === ""
+                        customHeight: 56
+                        anchors.leftMargin: 36
+                        anchors.top: parent.top
+                        textField.onActiveFocusChanged: {
+                            recipientInput.pristine = false
+                        }
+                        onTextChanged: {
+                            let isValidEns = Utils.isValidEns(text)
+                            let isValidAddress = Utils.isValidAddress(text)
+                            recipientInput.valid = isValidEns || isValidAddress
+
+                            if (!recipientInput.valid || text == "") {
+                                sendTransactionModal.recipient = null
+                                validationError = qsTr("This is an invalid address")
+                            } else {
+                                sendTransactionModal.recipient = {
+                                    address: text
+                                }
+                                validationError = ""
+                            }
+                        }
+
+                        Rectangle {
+                            width: 36
+                            height: 24
+                            visible: !recipientInput.pasteFromClipboard
+                            anchors.bottom: parent.bottom
+                            anchors.bottomMargin: checkmarkIcon.visible ? 16 : 33
+                            anchors.right: parent.right
+                            anchors.rightMargin: 6
+                            color: Theme.palette.baseColor2
+                            StatusIcon {
+                                id: cancelIcon
+                                visible: !recipientInput.valid && !recipientInput.pasteFromClipboard
+                                icon: "cancel"
+                                color: Theme.palette.dangerColor1
+                                height: 16
+                                anchors.centerIn: parent
+                                anchors.rightMargin: 8
+                            }
+                            StatusIcon {
+                                id: checkmarkIcon
+                                visible: recipientInput.valid && !recipientInput.pasteFromClipboard
+                                icon: "tiny/checkmark"
+                                height: 16
+                                color: Theme.palette.primaryColor1
+                                anchors.centerIn: parent
+                                anchors.rightMargin: 8
+                            }
+                        }
+                    }
```

Comment:
This an entire `Rectangle` is somewhat of a hack until `StatusInput` supports `component` as mentioned in: https://github.com/status-im/StatusQ/issues/380